examples/common/src/framework.rs

```diff
@@ -307,11 +307,20 @@ fn start<E: Example>(
                     },
                 ..
             } => {
-                log::info!("Resizing to {:?}", size);
-                config.width = size.width.max(1);
-                config.height = size.height.max(1);
-                example.resize(&config, &device, &queue);
-                surface.configure(&device, &config);
+                let max_dimension = adapter.limits().max_texture_dimension_2d;
+                if size.width > max_dimension || size.height > max_dimension {
```

Comment:
I don't get how this is happening in the first place. The limits for max texture dimension should be raised to the maximum supported by the adapter for this very reason.

Reply:
This seems to be a bug in winit itself. I have also encountered the same crash when running examples of bevy on my system.

Reply:
What is the dimension given by winit?

Reply:
u32::MAX, only happened on macOS Sonoma.

Reply:
The fuck? Can you file a bug with winit, then note that that bug is the reason for this?

Reply:
Sure.

Reply:
Someone already filed this bug two weeks ago: https://github.com/rust-windowing/winit/issues/2876

Reply:
Alright, if you can just add the reasoning to the code and then this should be good.